### ISSUE TITLE
Editor is taking longer than expected to clone repos

### ISSUE THREAD

**Issue by araknast** [inlang/inlang#902]
## Problem
In my experiments trying to pinpoint exactly what is slowing down cloning in the editor, I found that a non-insignificant amount of the loading time happens _after_ the repository is cloned, and after the config is read. From the time the loading spinner appears to the time the messages appear, about 2/3 of the loading time can be attributed to git. However, after the repository is cloned and the config is read, there are about 5 extra seconds of loading before the repository appears. (Measured when cloning `AppFlowy-IO/AppFlowy`)

### Expected behavior
Loading spinner disappears once the repo is cloned and config is loaded.

## Steps to reproduce
1. Add console.log lines in `source-code/website/src/pages/editor/@host/@owner/@repository/State.tsx` after the git.clone in `cloneRepository` and after the config is read (or clone https://github.com/araknast/inlang/tree/experiments)
2. `npm run dev`
3. navigate to http://localhost:3000
4. Open the console
5. Open AppFlowy-IO/AppFlowy in the editor

After the log line signifying that the config has been read, around 5 seconds pass before the loading spinner disappears and the messages appear.

### Additional information
- Browser: Firefox 111.0.1
- Operating system: Void Linux (glibc)

**Comment by samuelstroschein** [comment]
I suspect a state issue/and or rendering to be the cause.

https://github.com/inlang/inlang/issues/839 should speed up state things. If we can add the solid dev tools (https://github.com/inlang/inlang/issues/428), we might be able to investigate even deeper.

**Comment by araknast** [comment]
> I suspect a state issue/and or rendering to be the cause.
> 
> #839 should speed up state things. If we can add the solid dev tools (#428), we might be able to investigate even deeper.
> 
> @araknast PS I recommend always default assigning a person to an issue. Otherwise, the issue will not be worked on. Can you take this one over?

I've assigned myself, but I was hoping the rest of the team might have more insight as I'm still not super familiar with the inner workings of SolidJS. If not, then I'll take it over and hopefully it is fixed with the reactive fs.

**Comment by samuelstroschein** [comment]
Indeed 2/3 is repository related, and 1/3 is the parsing of resources. 

I stumbled on this by working on #937 where I realized that the plugin is highly inefficient. I conducted a performance analyzation, see this loom https://www.loom.com/share/96fc0e4827754000aafd91b84ea86be0. 

- network is by far the slowest thing. @araknast lazy cloning gotta come.
- followed by checkout (?)
- followed by parsing of resources 

I will close this issue in favor of #937
